src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = (function() {
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  function uniq(list) {
+    /* global Set */
+    var item,
+        set = new Set(),
+        idx = -1,
+        len = list.length,
+        items = [],
+        uniqs = [];
+
+    while (++idx < len) {
+      item = list[idx];
+      // `_contains` is also used to differentiate between
+      // +0 and -0, as the native Set does not.
+      if (item === 0 || typeof item.equals === 'function') {
```

Comment:
Do we need to replace

```javascript
typeof item.equals === 'function'
```

with

```javascript
item != null && typeof item.equals === 'function'
```

to avoid throwing a type error when `item` is null or undefined? If so, we should include a test for this.